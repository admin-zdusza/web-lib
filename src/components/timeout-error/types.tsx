export enum TIMEOUT {
  OCCURED = "TIMEOUT_OCCURED",
  ACKNOWLEDGED = "TIMEOUT_ACKNOWLEDGED",
}

export type TimeoutAction =
  | { type: TIMEOUT.OCCURED }
  | { type: TIMEOUT.ACKNOWLEDGED };
