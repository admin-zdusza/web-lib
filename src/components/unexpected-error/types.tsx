export enum UNEXPECTED_ERROR {
  OCCURED = "UNEXPECTED_ERROR_OCCURED",
  ACKNOWLEDGED = "UNEXPECTED_ERROR_ACKNOWLEDGED",
}

export type UnexpectedErrorAction =
  | { type: UNEXPECTED_ERROR.OCCURED }
  | { type: UNEXPECTED_ERROR.ACKNOWLEDGED };
