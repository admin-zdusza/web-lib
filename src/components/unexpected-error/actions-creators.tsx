import { UNEXPECTED_ERROR } from "./types";

const unexpectedErrorOccured = () => ({ type: UNEXPECTED_ERROR.OCCURED });

const unexpectedErrorAcknowledged = () => ({
  type: UNEXPECTED_ERROR.ACKNOWLEDGED,
});

export { unexpectedErrorOccured, unexpectedErrorAcknowledged };
