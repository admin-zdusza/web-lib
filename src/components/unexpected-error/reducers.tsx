import { UNEXPECTED_ERROR, UnexpectedErrorAction } from "./types";

const initialState: boolean = false;

const unexpectedErrorReducer = (
  state = initialState,
  action: UnexpectedErrorAction
): boolean => {
  switch (action.type) {
    case UNEXPECTED_ERROR.OCCURED:
      return true;
    case UNEXPECTED_ERROR.ACKNOWLEDGED:
      return false;
    default:
      return state;
  }
};

export { unexpectedErrorReducer };
