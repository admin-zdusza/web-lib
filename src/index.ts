export { Brand } from './components/brand/Brand';
export { TimeoutError } from './components/timeout-error/TimeoutError'
export { timeoutOccured } from './components/timeout-error/actions-creators';
export { timeoutReducer } from './components/timeout-error/reducers';
