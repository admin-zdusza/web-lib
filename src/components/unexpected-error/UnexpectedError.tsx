import React, { Component } from "react";
import Modal from "react-bootstrap/Modal";
import Button from "react-bootstrap/Button";
import { unexpectedErrorAcknowledged } from "./actions-creators";

export interface UnexpectedErrorProps {
  dispatch: Function;
  closable: boolean;
}

const onHide = (closable: boolean, dispatch: Function) => () => {
  if (closable) {
    dispatch(unexpectedErrorAcknowledged());
  }
};

const onClick = (dispatch: Function) => () =>
  dispatch(unexpectedErrorAcknowledged());

export class UnexpectedError extends Component<UnexpectedErrorProps> {
  render() {
    return (
      <Modal
        show={true}
        onHide={onHide(this.props.closable, this.props.dispatch)}
      >
        <Modal.Header closeButton={this.props.closable}>
          <Modal.Title className="text-danger">
            Nastąpił nieoczekiwany błąd.
          </Modal.Title>
        </Modal.Header>

        <Modal.Body className="text-danger">
          <p>Proszę spróbować jeszcze raz.</p>
        </Modal.Body>

        {this.props.closable ? (
          <Modal.Footer>
            <Button variant="secondary" onClick={onClick(this.props.dispatch)}>
              Ok
            </Button>
          </Modal.Footer>
        ) : null}
      </Modal>
    );
  }
}
