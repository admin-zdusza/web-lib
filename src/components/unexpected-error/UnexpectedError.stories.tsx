import * as React from 'react';
import { Meta, Story } from '@storybook/react/types-6-0';
import { UnexpectedError, UnexpectedErrorProps } from './UnexpectedError';

export default {
  title: 'UnexpectedError',
  component: UnexpectedError,
  description: `An unexpected error!`,
  argTypes: {},
} as Meta;

const Template: Story<UnexpectedErrorProps> = (args) => <UnexpectedError {...args}/>;

export const Closable = Template.bind({});
Closable.parameters = {
  viewMode: 'canvas',
  docs: {
    page: null
  }
};

Closable.args = {
  closable: true,
};

export const NonClosable = Template.bind({});
NonClosable.parameters = {
  viewMode: 'canvas',
  docs: {
    page: null
  }
};

NonClosable.args = {
  closable: false,
};
