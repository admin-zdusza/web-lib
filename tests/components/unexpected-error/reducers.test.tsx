import { unexpectedErrorReducer } from "../../../src";
import { UNEXPECTED_ERROR } from "../../../src/components/unexpected-error/types";

it("should reduce unexpected error occured", () => {
  expect(
    unexpectedErrorReducer(undefined, { type: UNEXPECTED_ERROR.OCCURED })
  ).toMatchSnapshot();
});

it("should reduce unexpected error acknowledged", () => {
  expect(
    unexpectedErrorReducer(undefined, { type: UNEXPECTED_ERROR.ACKNOWLEDGED })
  ).toMatchSnapshot();
});

it("should reduce to default", () => {
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  expect(unexpectedErrorReducer(false, { type: undefined })).toMatchSnapshot();
});
