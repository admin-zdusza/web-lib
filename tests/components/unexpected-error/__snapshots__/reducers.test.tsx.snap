// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should reduce to default 1`] = `false`;

exports[`should reduce unexpected error acknowledged 1`] = `false`;

exports[`should reduce unexpected error occured 1`] = `true`;
